controllers/core/openstackversion_controller.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	"github.com/openstack-k8s-operators/openstack-operator/pkg/openstack"
+)
+
+var envContainerImages (map[string]*string)
+var envAvailableVersion string
+
+// SetupVersionDefaults -
+func SetupVersionDefaults() {
+	fmt.Println("SetupVersionDefaults")
+	localVars := make(map[string]*string)
+	for _, name := range os.Environ() {
+		envArr := strings.Split(name, "=")
+		if envArr[0] == "OPENSTACK_RELEASE_VERSION" {
+			envAvailableVersion = envArr[1]
+		}
+		if strings.HasPrefix(envArr[0], "RELATED_IMAGE_") {
+			localVars[envArr[0]] = &envArr[1]
+		}
+	}
+	envContainerImages = localVars
+}
+
+func compareStringPointers(a, b *string) bool {
```

Comment:
It does not read nicely in the conditions below. As it returns a boolean I would call it `equalStringPointers` or `stringPointersEqual`instead